functions/query/query.go

```diff
@@ -161,16 +161,11 @@ func validateRequest(eventBody []byte) (*request, error) {
 	}
 
 	request := request{}
-	metric, ok := requestMap["metric"]
-	delete(requestMap, "metric")
-	if !ok {
-		return nil, errors.New("Request object is missing 'metric' field")
-	}
-	request.Metric, ok = metric.(string)
-	if !ok {
-		return nil, errors.New("'metric' field must be a string")
-	}
 	var err error
+	request.Metric, err = validateOptionalString(requestMap, "metric")
```

Comment:
We need to validate that we have `Metric` or `filter` but at least one of them is required